### PR TITLE
Add Event tracking for activate/acquire

### PR DESCRIPTION
**Pull request by fearthecowboy**

### CODE REVIEW

File: ce/insights.ts [hunk 1]
```diff
@@ -23,18 +25,50 @@ export const insights = setup('b4e88960-4393-4dd9-ab8e-97e8fe6d7603').
...
     tagOverrides: {
       'ai.device.os': process.platform,
       'ai.device.osVersion': version(),
     },
+
     properties: {
       ...properties,
-      'from': 'vcpkg-ce'
+      'kind': 'vcpkg-ce',
+      'ce.version': Version,
```
Comment (benmcmorran) [inline ce/insights.ts:60]: You can send this in `tagOverrides` using `defaultClient.context.keys.applicationVersion`.

**Review by benmcmorran — APPROVED** [review]

File: ce/insights.ts [hunk 2]
```diff
@@ -23,18 +25,50 @@ export const insights = setup('b4e88960-4393-4dd9-ab8e-97e8fe6d7603').
...
+defaultClient.addTelemetryProcessor((envelope, contextObjects) => {
+  if (session.context['printmetrics']) {
+    session.channels.message(`Telemetry Event: \n${JSON.stringify(envelope.data, null, 2)}`);
+  }
+
+  // only actually send telemetry if it's enabled.
+  return session.telemetryEnabled;
+});
+
+export function trackEvent(name: string, properties: { [key: string]: string } = {}) {
```
[the remark this answers is not in the document]
Comment (fearthecowboy) [reply ce/insights.ts:46]: Yeah, when we actually have some metrics to add. We just don't have any perf numbers at all for the moment.

File: ce/insights.ts (hunk 1 above)
Comment (fearthecowboy) [reply ce/insights.ts:60]: ok

File: ce/insights.ts [hunk 3]
```diff
@@ -23,18 +25,50 @@ export const insights = setup('b4e88960-4393-4dd9-ab8e-97e8fe6d7603').
...
+  session.channels.debug(`Triggering Telemetry Event ce.${name}`);
   defaultClient.trackEvent({
-    name: 'Microsoft.ApplicationInsights.Event',
+    name: `ce.${name}`,
```
[the remark this answers is not in the document]
Comment (fearthecowboy) [reply ce/insights.ts:49]: I was gonna do slash, but then looking thru github, I saw a lot of MS usage with dots. 

I spoke with them yesterday, and they are OK with adding a bunch of new event names. They're going to overhaul theirs at some point.

File: ce/cli/artifacts.ts [hunk 4]
```diff
@@ -120,6 +121,9 @@ export async function installArtifacts(artifacts: Iterable<Artifact>, options?:
       });
       // remember what was actually installed
       installed.set(artifact, actuallyInstalled);
+      if (actuallyInstalled) {
+        trackAcquire(artifact.id, artifact.version);
```
[the remark this answers is not in the document]
Comment (fearthecowboy) [reply ce/cli/artifacts.ts:125]: It's just the ids and versions at the moment, it doesn't even track where it's coming from. (or **what** it is.) -- the id is an arbitrary string (ie `tools/compilers/foo` ) ... 

Although now that you say it, I could hash those so we don't know what, but could correlate back with things that we did specify.

File: ce/insights.ts [hunk 5]
```diff
@@ -23,18 +25,50 @@ export const insights = setup('b4e88960-4393-4dd9-ab8e-97e8fe6d7603').
...
+    time: new Date(),
+
     tagOverrides: {
       'ai.device.os': process.platform,
       'ai.device.osVersion': version(),
```
[the remark this answers is not in the document]
Comment (fearthecowboy) [reply ce/insights.ts:54]: I really wish the docs would be clearer about usage of this stuff. 

I can't tell if I really have to set that or if the library was supposed to.

After walking thru the source for the appinsights library, it looks like they do set that already.

File: ce/cli/artifacts.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (benmcmorran) [reply ce/cli/artifacts.ts:125]: My concern is that if someone uses a private registry, the ID itself could be sensitive, like `libs/codenameformysupersecretproject`.

[the remark this answers is not in the document]
Comment (fearthecowboy) [reply ce/cli/artifacts.ts:125]: Yeah, which is why I'll hash those values.

[the remark this answers is not in the document]
Comment (fearthecowboy) [reply ce/cli/artifacts.ts:125]: I hash them in the function now

File: ce/insights.ts [hunk 6]
```diff
@@ -23,18 +25,47 @@ export const insights = setup('b4e88960-4393-4dd9-ab8e-97e8fe6d7603').
...
-    name: 'Microsoft.ApplicationInsights.Event',
-    tagOverrides: {
-      'ai.device.os': process.platform,
-      'ai.device.osVersion': version(),
-    },
+    name: `ce/${name}`,
+    time: new Date(),
+
     properties: {
       ...properties,
```
Comment (fearthecowboy) [inline ce/insights.ts:57]: cloning it to ensure that the properties object isn't shared between any event calls (which could happen in the future)

File: ce/insights.ts [hunk 7]
```diff
@@ -26,6 +26,9 @@ export const insights = setup('b4e88960-4393-4dd9-ab8e-97e8fe6d7603').
   setUseDiskRetryCaching(false).
   start();
 
+defaultClient.context.keys.applicationVersion = Version;
```
Comment (benmcmorran) [inline ce/insights.ts:29]: Sorry I guess I wasn't clear about this. `defaultClient.context.keys.applicationVersion` is [hardcoded to `ai.application.ver`](https://npm.runkit.com/applicationinsights/out/Declarations/Contracts/Generated/ContextTagKeys.js?t=1637782496370). I was saying that you could use it as the key in `tagOverrides` object.

**Review by benmcmorran — APPROVED** [review]

**Pull request merged**